### PR TITLE
Input: Change the default deadzone value for new actions from 0.5 to 0.2

### LINKED ISSUE #7069: Decrease the default deadzone value for actions to better match real world use cases
### Describe the project you are working on

A game allowing a gamepad as input.

### Describe the problem or limitation you are having in your project

The default deadzone value is `0.5`. By being the default value this implies that it's a reasonable choice.

[Other people recommend a much lower deadzone value (0.05)](https://github.com/godotengine/godot-proposals/issues/3709) and even the [Godot docs use 0.2 as an example value](https://docs.godotengine.org/en/stable/classes/class_inputmap.html#class-inputmap-method-add-action).
According to my experience this also behaves weirdly (on my Xbox 360 controller this feels more like using a d-pad than a stick). This might also confuse newer devs that have never heard about deadzones.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Change the default value to something that is more realistic but also works with older/worse hardware.
Maybe to the `0.2` mentioned in the Godot docs.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Change the default value.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Basically everyone must change the setting in the input map.

### Is there a reason why this should be core and not an add-on in the asset library?

It's already core, it's just about changing the default.

### PR DESCRIPTION
**Pull request by ghost**
This changes the default deadzone value for new actions from 0.5 to 0.2 as discussed in https://github.com/godotengine/godot-proposals/issues/7069.

---

I have failed to rebase my previous year old PR, (#80299) so I pressed the "Sync" button in the Github's UI, not knowing it what it would do 😊👍️
Turns out it erases all your changes with no warning and without any way to undo what you have done.

Please check the changes I've recreated and sorry for the mess.

*Bugsquad edit:*
- Closes https://github.com/godotengine/godot-proposals/issues/7069

### CODE REVIEW

File: core/input/input_map.cpp [hunk 1]
```diff
@@ -43,7 +43,7 @@ int InputMap::ALL_DEVICES = -1;
 void InputMap::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("has_action", "action"), &InputMap::has_action);
 	ClassDB::bind_method(D_METHOD("get_actions"), &InputMap::_get_actions);
-	ClassDB::bind_method(D_METHOD("add_action", "action", "deadzone"), &InputMap::add_action, DEFVAL(0.5f));
+	ClassDB::bind_method(D_METHOD("add_action", "action", "deadzone"), &InputMap::add_action, DEFVAL(0.2f));
```
Comment (AThousandShips) [inline core/input/input_map.cpp:46]: You need to add a compatibility method, see [here](https://docs.godotengine.org/en/latest/contributing/development/handling_compatibility_breakages.html) for instructions

Comment (reptofrog) [reply core/input/input_map.cpp:46]: @AThousandShips When I call `ClassDB::bind_compatibility_method`, in `_bind_compatibility_methods()` should I use the old DEFVAL value (`0.5`) or the new one? (`0.2`)

Comment (AThousandShips) [reply core/input/input_map.cpp:46]: The old one, to catch the old method

File: core/input/input_map.h [hunk 2]
```diff
@@ -66,6 +66,11 @@ class InputMap : public Object {
 	TypedArray<InputEvent> _action_get_events(const StringName &p_action);
 	TypedArray<StringName> _get_actions();
 
+#ifndef DISABLE_DEPRECATED
+	void _add_action_bind_compat_97281(const StringName &p_action, float p_deadzone);
```
Comment (AThousandShips) [inline core/input/input_map.h:70]:
```suggestion
	void _add_action_bind_compat_97281(const StringName &p_action, float p_deadzone = 0.5);
```
Just to make it clearer

Comment (AThousandShips) [reply core/input/input_map.h:70]: These should also be under `protected` not `private`

**Review by AThousandShips — APPROVED** [review]
Once those minor details are fixed this looks good to me!

**Review by Sauermann — COMMENTED** [review]
The reasoning for changing the default value, as discussed in the linked proposal, makes sense to me.

Can you please double check, if it would make sense to make additional adjustments to the following location in the code base:

https://github.com/godotengine/godot/blob/506d6e427a4eecfc1e5e5ee1180019a876119701/modules/gdscript/tests/scripts/project.godot#L15

**Comment by reptofrog** [comment]
I thought this could be left as it is, but I guess I could change that to the new default value (0.2) for the consistency's sake? I am not sure what to do with it.

**Review by Sauermann — APPROVED** [review]
From the input side, this looks good.

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Comment by wareya** [comment]
Did this PR only apply to new input actions, or did it apply to the built-in UI actions as well? Diagnostic posts in https://github.com/godotengine/godot/issues/98982 seem to imply that it applies to the built-in UI actions too, which would result in these changes causing this issue: https://github.com/godotengine/godot/issues/103360

**Comment by Calinou** [comment]
> Did this PR only apply to new input actions, or did it apply to the built-in UI actions as well? Diagnostic posts in #98982 seem to imply that it applies to the built-in UI actions too, which would result in these changes causing this issue: #103360

It applies to built-in actions too, as these are often used by people for gameplay logic during prototyping.

Built-in actions are not saved to `project.godot` unless you override them, so if we change it back to `0.5`, it shouldn't affect custom actions.

Ideally, I think we should use custom deadzone handling for UI logic where it always uses a separate, fixed deadzone (e.g. as a project setting). This may be difficult to do though, so maybe reverting the deadzone for built-in actions only is preferable, even if it hurts the prototyping experience for users who don't create new actions (or people using the CharacterBody script template).

**Pull request merged**